ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionLogUpdatesConversions.scala

```diff
@@ -182,43 +194,40 @@ private[events] object TransactionLogUpdatesConversions {
             eventId = exercisedEvent.eventId.toLedgerString,
             contractId = exercisedEvent.contractId.coid,
             templateId = Some(LfEngineToApi.toApiIdentifier(exercisedEvent.templateId)),
-            witnessParties = requestingParties.view.filter(exercisedEvent.flatEventWitnesses).toSeq,
+            witnessParties =
+              requestingParties.iterator.filter(exercisedEvent.flatEventWitnesses).toSeq,
           )
         )
       )
   }
 
   object ToTransactionTree {
-    def apply(
-        tx: TransactionLogUpdate.Transaction,
-        requestingParties: Set[String],
+    def filter(requestingParties: Set[Party]): Filter = transaction => {
+      val filteredForVisibility =
+        transaction.events
+          .filter(transactionTreePredicate(requestingParties))
+
+      Option.when(filteredForVisibility.nonEmpty)(transaction.copy(events = filteredForVisibility))
+    }
+
+    def toApiTransaction(
+        requestingParties: Set[Party],
         verbose: Boolean,
         lfValueTranslation: LfValueTranslation,
     )(implicit
         loggingContext: LoggingContext,
         executionContext: ExecutionContext,
-    ): Future[Option[TransactionTree]] = {
-      val filteredForVisibility = tx.events
-        .filter(TransactionTreePredicate(requestingParties))
-
-      if (filteredForVisibility.isEmpty) Future.successful(None)
-      else
+    ): ToApi[GetTransactionTreesResponse] =
+      transaction =>
         Future
-          .traverse(filteredForVisibility)(
+          .traverse(transaction.events)(
             toTransactionTreeEvent(requestingParties, verbose, lfValueTranslation)
           )
           .map { treeEvents =>
             val visible = treeEvents.map(_.eventId)
-            val visibleOrder = visible.view.zipWithIndex.toMap
+            val visibleSet = visible.toSet
             val eventsById = treeEvents.iterator
-              .map(e =>
-                e.eventId -> e
-                  .filterChildEventIds(visibleOrder.contains)
-                  // childEventIds need to be returned in the event order in the original transaction.
-                  // Unfortunately, we did not store them ordered in the past so we have to sort it to recover this order.
-                  // The order is determined by the order of the events, which follows the event order of the original transaction.
-                  .sortChildEventIdsBy(visibleOrder)
```

Comment:
Why do you think you can remove this? Also changes above: you seem like rolling back my changes to fix child event id sot order. Is this a copy paste issue? Or will this be done someplace else?

Reply:
No, this was not intentional, I suspect this might be due to a rebase. However, why didn't the newly introduced test fail?

Reply:
Which makes me realize: We don't actually need this fix here. The in-memory fan-out buffers cannot be populated with data ingested by an older process (which might have run the indexer that was not sorting these).

Reply:
yes, this "double fixing" has the same issues with the normal persistent TransactionReader path as well. Conformance tests only uncover problems if the the indexer fix is manually reverted.
On your offline comment (whether this fix for the IMFO is needed at all, since IMFO only targets new data, which is stored already correctly with the indexer fix): you are generally right, but there could be corner cases with split-participant deployment, so for consistency on main, I would only remove the fix for this part, if split participant feature is removed.

Reply:
after IMFO is fed from indexer directly (not from DB) this can be removed. (TODO)